Data/ByteString/Lazy/Char8.hs

```diff
@@ -399,6 +409,38 @@ minimum = w2c . L.minimum
 scanl :: (Char -> Char -> Char) -> Char -> ByteString -> ByteString
 scanl f z = L.scanl (\a b -> c2w (f (w2c a) (w2c b))) (c2w z)
 
+-- | 'scanl1' is a variant of 'scanl' that has no starting value argument.
+-- This function will fuse.
+--
+-- > scanl1 f [x1, x2, ...] == [x1, x1 `f` x2, ...]
+scanl1 :: (Char -> Char -> Char) -> ByteString -> ByteString
+scanl1 f = L.scanl1 f' where f' accumulator value = c2w (f (w2c accumulator) (w2c value))
```

Comment:
If you want to fit all in one line, I'd rather use `Data.Function.on`: 
```
scanl1 f = L.scanl1 ((c2w .) . (f `on` w2c))
```
Otherwise I suggest we put `where` on a separate line to be more consistent with other code.

Reply:
I am not sure you are right about consistency with other code. It is really not consistent already. There are two places elsewhere that do this:

https://github.com/haskell/bytestring/blob/b639596ef7913b8cffd1cd028649a30ec038f1c5/Data/ByteString/Lazy.hs#L631-L631
https://github.com/haskell/bytestring/blob/b639596ef7913b8cffd1cd028649a30ec038f1c5/Data/ByteString/Lazy.hs#L654-L654

I am going to put a line break before `where`, and count these examples as pathologic.